src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala

```diff
@@ -82,7 +82,15 @@ trait LibraryServiceSupport extends LazyLogging {
       case None => Map()
     }
 
-    val fields = attrfields ++ idfields ++ tagfields
+    val dur: Map[AttributeName, Attribute] = generateStructuredUseRestriction(workspace)
+
+    val fields = dur match {
+      case x if x.isEmpty => attrfields ++ idfields ++ tagfields
+      case _ =>
+        // remove the data use attributes so we don't index them twice
+        val durAttributeNames = durFieldNames.map(AttributeName.withLibraryNS)
+        attrfields.filterKeys { key => !durAttributeNames.contains(key) } ++ idfields ++ tagfields ++ dur
+    }
```

Comment:
`filterKeys` is unsafe in the same way `mapValues` is: https://issues.scala-lang.org/browse/SI-4776. I'm guessing the use of `filterKeys` here is okay, but I'd like to keep it out of the codebase altogether for sanity's sake. With that in mind, I think you can simplify this to:

```
(attrfields -- durAttributeNames) ++ idfields ++ tagfields ++ dur
```

Also - I don't see any reason to index the consent codes at all, if `dur` is empty. If we don't have a valid dur, there's no point in indexing them, since we can't rely on their values, and we don't have any code that looks for them outside of the dur. I say remove the match/case and always use the second codepath.

Reply:
Argh ... yes, I should have remembered the filterKeys issue. Will fix.